src/go/rpk/pkg/cli/redpanda/config.go

```diff
@@ -173,6 +192,13 @@ you must use the --self flag to specify which ip redpanda should listen on.
 					Port:    config.DefaultKafkaPort,
 				}}
 			}
+			if a := &y.Redpanda.AdvertisedKafkaAPI; len(*a) == 1 {
```

Comment:
Does this need to be >= 1?

Reply:
The `==1` is intentional:
https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/redpanda/config.go#L142-L147

> 			// Defaults returns one RPC, one KafkaAPI, and one
>			// AdminAPI. We only override values in a configuration
>			// file if the current values are defaults, or if an
>			// address array is empty. If an address array has
>			// multiple elements, we trust that user modifications
>			// were intentional.